docs/usage/dns-search-path-optimization.md

```diff
@@ -0,0 +1,113 @@
+
+## DNS Search Path/`ndots` Problem in Kubernetes
+
+As the DNS search path is long and `ndots` is large, a lot of DNS queries might traverse the DNS search path. This results
+in an explosion of DNS requests.
+
+For example, consider the name resolution of the default kubernetes service `kubernetes.default.svc.cluster.local`. As this
+name has only four dots it is not considered a fully qualified name according to the default `ndots=5` setting. Therefore,
+the DNS search path is applied resulting in the following queries being created
+ - `kubernetes.default.svc.cluster.local.some-namespace.svc.cluster.local`
+ - `kubernetes.default.svc.cluster.local.svc.cluster.local`
+ - `kubernetes.default.svc.cluster.local.cluster.local`
+ - `kubernetes.default.svc.cluster.local.network-domain`
+ - ...
+
+In IPv4/IPv6 dual stack systems, the amount of DNS requests may even double as each name is resolved for IPv4 and IPv6.
+
+## General Workarounds/Mitigations
+
+Kubernetes provides the capability to set the DNS options for each pod (see
+[Pod DNS config](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config) for details).
+However, this has to be applied for every pod (doing name resolution) to resolve the problem. A mutating webhook may be
+useful in this regard. Unfortunately, the DNS requirements may be different depending on the workload. Therefore, a general
+solution may difficult to impossible.
+
+Another approach is to use always fully qualified names and append a dot (`.`) to the name to prevent the name resolution
+system from using the DNS search path. This might be somewhat counterintuitive as most developers are not used to the
+trailing dot (`.`). Furthermore, it makes moving to different landscapes more difficult/error-prone.
+
+## Gardener specific Workarounds/Mitigations
+
+Gardener allows users to [customize their DNS configuration](custom-dns.md). CoreDNS allows several approaches to deal with
+the requests generated by the DNS search path. [Caching](https://coredns.io/plugins/cache/) is possible as well as
+[query rewriting](https://coredns.io/plugins/rewrite/). There are also several other [plugins](https://coredns.io/plugins/)
+available, which may mitigate the situation.
+
+## Gardener DNS Query Rewriting
+
+As explained [above](#dns-search-path-ndots-and-kubernetes), the application of the DNS search path may lead to the undesired
+creation of DNS requests. Especially with the default setting of `ndots=5`, seemingly fully qualified names pointing to
+services in the cluster may trigger the DNS search path application.
+
+Gardener allows to automatically rewrite some obviously incorrect DNS names, which stem from application of the DNS search
+path, to the most likely desired name. The feature can be enabled by setting the Gardenlet feature-gate `CoreDNSQueryRewriting` to `true`:
```

Comment:
Does it also make sense to consider an annotation to disable the feature per Shoot (similar to what we had for node-local-dns in the past)? Or will it be added only if there is a need for it?
Motivation: it can be helpful to be able to disable per Shoot basis if the Shoot experiences issues after the feature enablement.

Reply:
The initial pull request had the `enabled` flag directly in the shoot specification. However, it was decided to remove it. Adding the annotation feels a bit like reintroducing it.
However, I have no objection and added this in the latest commit.